components/script/indexed_db.rs

```diff
@@ -35,9 +38,16 @@ pub fn key_type_to_jsval(
         IndexedDBKeyType::Number(n) => result.set(DoubleValue(*n)),
         IndexedDBKeyType::String(s) => s.safe_to_jsval(cx, result),
         IndexedDBKeyType::Binary(b) => b.safe_to_jsval(cx, result),
-        IndexedDBKeyType::Date(_d) => {
-            // TODO: implement this when Date's representation is finalized.
-            result.set(UndefinedValue());
+        IndexedDBKeyType::Date(d) => unsafe {
+            // TODO: This doesn't clip the time (while firefox does)
+            // Inner representation of clippedtime is a f64, this is safe
+            let time: ClippedTime = std::mem::transmute(d.trunc());
```

Comment:
We should add a function to jsglue.cpp in mozjs that calls https://github.com/servo/mozjs/blob/6a966a78b63395fdd5755553d0c0ce79ef26f9e1/mozjs-sys/mozjs/js/public/Date.h#L102 and call it here instead. Transmutes like this are a source of technical debt for future mozjs upgrades.